src/k8s/cmd/k8s/k8s_kubectl.go

```diff
@@ -0,0 +1,49 @@
+package k8s
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/canonical/k8s/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	kubectlCmd = &cobra.Command{
+		Use:   "kubectl",
+		Short: "Integrated Kubernetes CLI",
+		// All commands should be passed to kubectl
+		DisableFlagParsing: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if rootCmdOpts.logDebug {
+				logrus.SetLevel(logrus.TraceLevel)
+			}
+
+			// Allow users to provide their own kubeconfig but
+			// fallback to the admin config if nothing is provided.
+			if os.Getenv("KUBECONFIG") == "" {
+				os.Setenv("KUBECONFIG", utils.SnapCommonPath("/etc/kubernetes/admin.conf"))
+			}
+			command := append(
+				[]string{utils.SnapPath("bin/kubectl"),
+					"--kubeconfig",
+					os.Getenv("KUBECONFIG"),
+				},
+				args...,
+			)
+
+			err := utils.RunCommand(cmd.Context(), command...)
+			if err != nil {
+				fmt.Println(err)
+				return fmt.Errorf("failed to execute kubectl command: %w", err)
+			}
```

Comment:
Do we have a reason to mess with the kubectl output? I would prefer just creating a `exec.Command`, hook up stdout and stderr then run and return the status code.

Reply:
It's OK if we don't use a `util` that does not really fit our purpose here, this is a single one-off wrapper command.

Reply:
Can you check if we could instead use https://pkg.go.dev/syscall#Exec, which handles everything for us, and ensures we are as close to a proper `kubectl` invocation as possible?

Reply:
Thanks for the recommendation - this worked!